docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+Nach dem obigen Beispiel würde dem Konsumenten-Dogu zwei Registry-Einträge einrichten, die das jeweilige Zugangsdatum als Wert enthalten:  
+
+- `/config/<konsument>/sa-<produzent>/username`
+- `/config/<konsument>/sa-<produzent>/passwort` 
+
+Das Verbaucher-Dogu kann diese nun z. B. durch `doguctl config -e sa-<produzentdogu>/username` auslesen und entschlüsseln.
+
+#### Service Account löschen
+
+Das `service-account-remove`-Skript löscht einen bestehenden Service Account eines Konsumenten-Dogus.
+
+Es wird immer ausgeführt, wenn das Konsumenten-Dogu deinstalliert wird. Während des Aufrufs erhält das `service-account-remove`-Skript jedenfalls den Namen des Konsumenten-Dogus als erstes Argument. Da hier keine Ausgaben des Skriptes verarbeitet werden, können auch Fehlermeldungen auf `stdout` ausgegeben werden.
+
+Ein `service-account-remove`-Skript besteht aus zwei Schritten:
+
+1. Identifier von dem Konsumenten identifizieren 
+2. Löschen der Daten und des Service Accounts des Konsumenten
+
+Ein Beispiel für das `service-account-remove`-Skript könnte folgendermaßen aussehen:
+
+`remove-sa.sh`
+
+```bash
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#1) Identifier von dem Konsumenten identifizieren
+SERVICE="${1}"
+if [ X"${SERVICE}" = X"" ]; then
+  echo "usage remove-sa.sh servicename"
+  exit 1
+fi
+
+#2) Löschen des Service Accounts des Verbrauchers
+# ... Implementiere die Dogu-spezifische Logik hier
+```
+
+In Schritt 1 wird der Name des Konsumenten-Dogus ausgelesen (z. B. "redmine"), i. d. R. um den Service Account des Dogus zu identifizieren.
+
+In Schritt 2 wird der Account aus der Datenbasis des Produzenten inkl. Zugangsdaten gelöscht.
+
+### Service Accounts konsumieren
+
+Es ist sehr einfach, bei einem Produzenten-Dogu einen Service Account anzufragen, da die Hauptarbeit beim Client und beim Produzenten liegt. Ein Konsumenten-Dogu muss lediglich in seiner `dogu.json` das gewünschte Produzenten-Dogu als [ServiceAccount]() nennen:
+
+```json
+{
+  ...,
+  "ServiceAccounts": [
+    {
+      "Type": "produzentendogu",
+      "Kind": "dogu"
+    }
+  ],
+  ...
+}
+```
+
+Bei einem Service Account handelt es sich um eine besondere Form der Abhängigkeit. Trotzdem ist es sinnvoll, das Produzenten-Dogu in seiner [Dependency-Liste]() zu führen. So wird in den unterschiedlichen Phasen einer Dogu-Installation sichergestellt, dass das Produzenten-Dogu wirklich zur Benutzung bereitsteht:
+
+
+```json
+{
+  ...,
+  "Dependencies": [
+    {
+      "name": "produzentendogu",
+      "type": "dogu"
+    }
+  ],
+  "ServiceAccounts": [
+    {
+      "Type": "produzentendogu",
+      "Kind": "dogu"
+    }
+  ],
+  ...
+}
+```
+
+## Dogu-Upgrades
+
+Das Upgrade eines Dogus kann mit `cesapp upgrade dogu <doguname>` durchgeführt werden. Dabei verwaltet die cesapp alle wichtigen Schritte, um das Dogu erfolgreich zu upgraden.
```

Comment:
I would like to abstract here from `cesapp`, since there is a K8s-centric CES solution on the horizon. We use to speak from "clients" that install, upgrade or delete dogus.

Reply:
resolved by fc6c870ef6216fda1079f061f8049a204eaf2c6e